interfaces/builtin/mir.go

```diff
@@ -74,6 +74,7 @@ var mirConnectedPlugAppArmor = []byte(`
 unix (receive, send) type=seqpacket addr=none peer=(label=###SLOT_SECURITY_TAGS###),
 /run/mir_socket rw,
 /run/user/[0-9]*/mir_socket rw,
+/run/user/[0-9]*/snap.###SLOT_NAME###/mir_socket rw,
```

Comment:
Why isn't the unity8-session snap using `/run/user/[0-9]*/mir_socket`? What this rule seems to be saying is that the location of the mir socket will vary based on the slot implementation, which is a tight coupling between a specific slot implementation and the snap and the dependency required to make this tight coupling work cannot be expressed in snap/snapcraft.yaml. More concretely, this means that the slot's socket is in /run/user/1000/snap.unity8-session/mir_socket and a unity8 plugging app would need to look for it there. An alternate implementation of unity8 would be at /run/user/1000/snap.unity8-session-somethingelse/mir_socket and all snaps that use 'plugs: [ unity8 ]' will not be able to find it.

The purpose of interfaces is that they are a contract between the slot side and the plug side such that a plug doesn't have to care what slot snap it is connecting to and vice versa.